x/recovery/keeper/ibc_callbacks_integration_test.go

```diff
@@ -0,0 +1,435 @@
+	params.EnableRecovery = false
+	suite.EvmosChain.App.(*app.Evmos).RecoveryKeeper.SetParams(suite.EvmosChain.GetContext(), params)
+
+	sender := suite.IBCOsmosisChain.SenderAccount.GetAddress().String()
+	receiver := suite.IBCOsmosisChain.SenderAccount.GetAddress().String()
+
+	pathOsmosisEvmos := suite.pathOsmosisEvmos
+	pathOsmosisCosmos := suite.pathOsmosisCosmos
+
+	suite.SendAndReceiveMessage(pathOsmosisCosmos, suite.IBCCosmosChain, "uatom", 10, suite.IBCCosmosChain.SenderAccount.GetAddress().String(), receiver, 1)
+
+	// Send IBC transaction of 10 uosmo
+	transferMsg := transfertypes.NewMsgTransfer(pathOsmosisEvmos.EndpointA.ChannelConfig.PortID, pathOsmosisEvmos.EndpointA.ChannelID, sdk.NewCoin(uatomIbcdenom, sdk.NewInt(10)), sender, receiver, timeoutHeight, 0)
+	_, err := suite.IBCOsmosisChain.SendMsgs(transferMsg)
+	suite.Require().NoError(err) // message committed
+	transfer := transfertypes.NewFungibleTokenPacketData("transfer/channel-1/uatom", "10", sender, receiver)
+	packet := channeltypes.NewPacket(transfer.GetBytes(), 1, pathOsmosisEvmos.EndpointA.ChannelConfig.PortID, pathOsmosisEvmos.EndpointA.ChannelID, pathOsmosisEvmos.EndpointB.ChannelConfig.PortID, pathOsmosisEvmos.EndpointB.ChannelID, timeoutHeight, 0)
+	// Receive message on the evmos side, and send ack
+	err = pathOsmosisEvmos.RelayPacket(packet)
+	suite.Require().NoError(err)
+
+	params.EnableRecovery = true
+	suite.EvmosChain.App.(*app.Evmos).RecoveryKeeper.SetParams(suite.EvmosChain.GetContext(), params)
+
+	suite.SendAndReceiveMessage(pathOsmosisEvmos, suite.IBCOsmosisChain, "uosmo", 10, sender, receiver, 2)
+	timeout := uint64(suite.EvmosChain.GetContext().BlockTime().Add(time.Hour * 4).Add(time.Second * -20).UnixNano())
+
+	// Recreate packets that were sent in the ibc_callback
+	// Coins locked
+	packet2 := CreatePacket("10000", "aevmos", sender, receiver,
+		"transfer", "channel-0", "transfer", "channel-0", 1, timeout)
+
+	packet3 := CreatePacket("10", "transfer/channel-0/uosmo", sender, receiver,
+		"transfer", "channel-0", "transfer", "channel-0", 3, timeout)
+
+	packet4 := CreatePacket("10", "transfer/channel-0/transfer/channel-1/uatom", sender, receiver,
+		"transfer", "channel-0", "transfer", "channel-0", 2, timeout)
+
+	// Relay packets that were sent in the ibc_callback
+	err = pathOsmosisEvmos.RelayPacket(packet2)
+	suite.Require().NoError(err)
+	err = pathOsmosisEvmos.RelayPacket(packet3)
+	suite.Require().NoError(err)
+	err = pathOsmosisEvmos.RelayPacket(packet4)
+	suite.Require().NoError(err)
+
+	senderAcc, err := sdk.AccAddressFromBech32(sender)
+	suite.Require().NoError(err)
+	receiverAcc, err := sdk.AccAddressFromBech32(receiver)
+	suite.Require().NoError(err)
+
+	// Aevmos was recovered from user address
+	coin := suite.EvmosChain.App.(*app.Evmos).BankKeeper.GetBalance(suite.EvmosChain.GetContext(), senderAcc, "aevmos")
+	suite.Require().Equal(coin, sdk.NewCoin("aevmos", sdk.NewInt(0)))
+
+	// Check that the uosmo were retrieved
```

Comment:
we need to check that the ibcuatom was retrieved to osmosis

Reply:
add the check